### PR TITLE
Reset revision filter when switching repos

### LINKED ISSUE #3153: Clear "file filter" when I switch repository
I'd like the "file filter" to turn off automatically when I switch repository, because I think it makes sense from a usability point of view. To reproduce my frustration:

1) Cntrl+F
2) Enter a valid file filter and tick it on, to view just changes to those files
3) In the bar at the top, switch to a different repository
4) Your whole view is now empty because despite the directory/file not being found in the new repository, the filter is still active

Preferably, the setting should be stored with the repository, so that I can have one filter active on one repository and still use another repository without losing that information in the first repository I was viewing. An alternative would be to check if the file/folder exists after switching, turning it off if it doesn't match. It would also be fine to just turn off the tick whenever you switch repository.

A similar approach would be nice for branches, where I don't want to filter by a branch that doesn't exist, and would prefer it to be stored with the repository.

### PR DESCRIPTION
**Pull request by RussKie**
Closes #3153

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitUI/CommandsDialogs/FormBrowse.cs [hunk 1]
```diff
@@ -1978,6 +1978,7 @@ private void SetGitModule(object sender, GitModuleEventArgs e)
             _gitStatusMonitor.InvalidateGitWorkingDirectoryStatus();
             _submoduleStatusProvider.Init();
             ToolStripFilters.SetBranchFilter(string.Empty, refresh: false);
+            ToolStripFilters.SetRevisionFilter(string.Empty);
```
Comment (gerhardol) [inline GitUI/CommandsDialogs/FormBrowse.cs:1981]: Note for other reviewers: revision filters can be set at FormBrowse startup (as command lind arguments), but SetGitModule() does not execute at startup so filters are not cleared.

**Review by gerhardol — COMMENTED** [review]
The issue reports on the Advanced filter, paths, but the PR handles the toolbar revision filter. The revision is also relevant though.

However, I have made the path filter change in #9445, will adopt that to all items and push to this branch.

File: GitUI/UserControls/RevisionGrid/RevisionGridControl.cs [hunk 2]
```diff
@@ -409,7 +409,12 @@ private static ArgumentString FindRenamesAndCopiesOpts()
             return FindRenamesOpt() + findCopies;
         }
 
-        public void SetPathFilters(string path)
+        public void DisableFilters()
+        {
+            _formRevisionFilter.DisableFilters();
+        }
+
+        public void SetPathFilter(string path)
```
Comment (RussKie) [inline GitUI/UserControls/RevisionGrid/RevisionGridControl.cs:417]: We don't use "disable/clear" for other filters, instead we set those to `""`. Any reason you thought it wouldn't work in this case?

**Comment by RussKie** [comment]
Thank you, I was mostly oblivious about other filter dialogs 👍

**Pull request merged**